### PR TITLE
Itertools count

### PR DESCRIPTION
**Pull request by Algebra8**
In this PR, functionality and tests for `itertools.count` is included and is related to #8.

### CODE REVIEW

**Comment by Algebra8** [comment]
Hello @tdakkota! This is a really cool project and I'm glad to contribute!

It seems like Coverage is breaking and I think it might have to do with the fact that I'm making a PR to your master branch. Is that correct?

It seems like the issue is:

```
Codecov 1.0.2 checksums for SHA1 failed to match.
Public checksum:   537069158a6f72b145cfe5f782dceb608d9ef594  codecov
af0dd19ee59f977bf5793cffafe116d7c248aa62  env
Uploader checksum: 537069158a6f72b145cfe5f782dceb608d9ef594  codecov
```

I don't have any experience with codecov so I'm not sure how to resolve this issue and any help would be appreciated. Thanks!

**Comment by tdakkota** [comment]
Hi @Algebra8! Thank you for your contribution!

File: itertools.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package sun
+
+import (
+	"fmt"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+var ItertoolsModule = &starlarkstruct.Module{
```
Comment (tdakkota) [inline itertools.go:10]: I think `ItertoolsModule` should be in the separated package.

**Review by tdakkota — CHANGES_REQUESTED** [review]

File: itertools.go (hunk 1 above)
Comment (Algebra8) [reply itertools.go:10]: Ah ok, like put it in `module.go`? That makes sense, thanks for pointing that out!

Comment (Algebra8) [reply itertools.go]: I moved `ItertoolsModule` to `module.go` in [this commit](https://github.com/tdakkota/sun/pull/10/commits/e4e64551320a98eedb2cd2f80b1c651c1374b80d).

**Comment by tdakkota** [comment]
Thanks!

**Review by tdakkota — APPROVED** [review]

**Comment by Algebra8** [comment]
Thanks for merging!

A few  things:

1. Python's `itertools.count()` actually accepts floats as well as ints, but my implementation only accepts ints. I am working on a PR to handle that.
2. I have a local implementation of `itertools.islice()` that I will make a PR for after 1_ is complete.
3. I was under the assumption that I would become a contributor but it does not look like I am. Do you know why that could be the case:thinking:?

Thanks!

**Comment by tdakkota** [comment]
1. Great! It would be nice to add test to be sure that your implementation handles `NaN` like as python's
```python
>>> [x for x in zip(range(5), itertools.count(0.0, float('nan')))]
[(0, 0.0), (1, nan), (2, nan), (3, nan), (4, nan)]
>>> [x for x in zip(range(5), itertools.count(float('nan')))]
[(0, nan), (1, nan), (2, nan), (3, nan), (4, nan)]
```
2. Thanks!
3. I don't know. Usually, GitHub adds a `contributor` tag, if default branch contains your commits. Seems like bug.

**Comment by Algebra8** [comment]
So, there is an interesting issue here. For some context, consider the following output of Python's `itertools.count`:

```bash
>>> itertools.count(5, 0.5)
count(5, 0.5)
```

Here, to get the next value we would have to add an int, `5`, to a float, `0.5`. But as far as I can tell `starlark` does not have addition support between `starlark.Int` and `starlark.Float`.

So far, it seems to me that there are two things we can do:

1. Create custom types that extend `starlark`'s `Int` and `Float` types and implement this addition support. E.g.:

```go
type sunInt { starlark.Int }
type sunFloat starlark.Float

type number interface {
	add(starlark.Value) (starlark.Value, error)
	string() string
}

// Each of these functions would have a switch statement on v.(type) and handle accordingly
func (i sunInt) add(v starlark.Value) (starlark.Value, error) {...}
func (i sunFloat) add(v starlark.Value) (starlark.Value, error) {...}
```

2. We can use the `floatOrInt` as a `starlark.Unpacker` recipe to unpack `starlark.Int` and `starlark.Float` into a single var ([as can be seen in the math module](https://github.com/google/starlark-go/blob/master/lib/math/math.go#L113)). However, note that `floatOrInt` is essentially a Golang `float64` and this will be reflected in `countObject.String()`. That is:

```bash
>>> itertools.count(5, 0.5)
count(5.0, 0.5)
>>> itertools.count(2, 1)
count(2.0, 1.0)
```
And apparently `starlark` does not seem to mind this. If you look at their [`math` test data](https://github.com/google/starlark-go/blob/master/starlark/testdata/math.star#L17), you will notice:

```python
assert.eq(math.ceil(0), 0.0)
assert.eq(math.ceil(1), 1.0)
assert.eq(math.ceil(10), 10.0)
```

Even though Python's does not output a float from an int input:
```bash
>>> import math
>>> o = math.ceil(0)
>>> o
0
>>> type(o)
<class 'int'>
```
@tdakkota what do you think? 1 or 2, or is there a better way?

Thanks!

PS Both 1 and 2 should handle `+inf`, `-inf`, `nan`.

**Pull request merged**